### PR TITLE
fix: on touch progress dialog goes no where

### PR DESCRIPTION
**Pull request by DravitLochan**
**Summary of changes**

while a request is made, progress dialog comes up. Now, even on touch the dialog remains there. 

Please make sure these boxes are checked before submitting your pull request - thanks! [Guide](./PULL_REQUEST_GUIDE.md)

- [x] Code is formatted.

- [x] Run the lint tests with `./gradlew lintDebug` and make sure BUILD is SUCCESSFUL

- [x] If new feature is implemeted then it is compatible with Night theme too.

- [x] Commit messages are well-written.

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@DravitLochan, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by kunall17** [comment]
@DravitLochan the changes you made will make changes in all the places where this CommonDialog is used, are you sure we want to disable the cancelable in all the places? 

Also format your code before sending!

**Comment by kunall17** [comment]
Also have a look on commit messages guidlines https://zulip.readthedocs.io/en/latest/version-control.html#commit-messages

**Comment by DravitLochan** [comment]
ya I think it will be a bug everywhere. suppose a user by mistake touches on the screen and has a poor internet connection. then it won't be a good light

**Comment by DravitLochan** [comment]
@kunall17 anything changes you recommend?

**Comment by kunall17** [comment]
Yeah, 

* Firstly format your code using the reformat code in android studio 
* I would suggest to keep the dialog as default as possible, like wherever .show() is being called we should there set its not cancellable (Either by passing variable or calling a method)

**Comment by nitish1211** [comment]
Great work @DravitLochan 👍   . One more thing you could add here is making the progressDialog non-cancellable even during logout and Refresh of feed.

**Comment by DravitLochan** [comment]
setCancelable(false) has been removed from default. only at refresh, signin, logout and while connecting to server.

**Comment by DravitLochan** [comment]
@kunall17 @nitish1211  please review!

File: app/src/main/java/com/zulip/android/util/CommonProgressDialog.java [hunk 1]
```diff
@@ -15,6 +15,7 @@
     public CommonProgressDialog(Context context) {
         progressDialog = new ProgressDialog(context);
         progressDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
+        setCancelable(false);
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/util/CommonProgressDialog.java]: This change to be removed!

File: app/src/main/java/com/zulip/android/util/CommonProgressDialog.java [hunk 2]
```diff
@@ -37,4 +38,9 @@ public boolean isShowing() {
     public void setMessage(String message) {
         progressDialog.setMessage(message);
     }
+
+    public void setCancelable(boolean cancelable)
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/util/CommonProgressDialog.java]: Format the code here

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 3]
```diff
@@ -2137,6 +2138,7 @@ private void setNightMode(@AppCompatDelegate.NightMode int nightMode) {
      */
     private void logout() {
         commonProgressDialog.showWithMessage(getString(R.string.logging_out));
+        commonProgressDialog.setCancelable(false);
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: setCancelable here has no effect as you are calling after showWithMessage

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 4]
```diff
@@ -2081,6 +2081,7 @@ public void onClick(
             case R.id.refresh:
                 Log.w("menu", "Refreshed manually by user. We shouldn't need this.");
                 commonProgressDialog.showWithMessage(getString(R.string.refreshing));
+                commonProgressDialog.setCancelable(false);
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: setCancelable here has no effect as you are calling after showWithMessage

File: app/src/main/java/com/zulip/android/activities/LoginActivity.java [hunk 5]
```diff
@@ -559,6 +562,7 @@ public void onFailure(Call<LoginResponse> call, Throwable t) {
             case R.id.local_server_button:
                 if (!isInputValidForDevAuth()) return;
                 commonProgressDialog.showWithMessage(getString(R.string.signing_in));
+                commonProgressDialog.setCancelable(false);
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/LoginActivity.java]: setCancelable here has no effect as you are calling after showWithMessage

File: app/src/main/java/com/zulip/android/activities/LoginActivity.java [hunk 6]
```diff
@@ -498,13 +499,15 @@ public void onClick(View v) {
         switch (v.getId()) {
             case R.id.google_sign_in_button:
                 commonProgressDialog.showWithMessage(getString(R.string.signing_in));
+                commonProgressDialog.setCancelable(false);
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/LoginActivity.java]: setCancelable here has no effect as you are calling after showWithMessage

File: app/src/main/java/com/zulip/android/activities/LoginActivity.java [hunk 7]
```diff
@@ -498,13 +499,15 @@ public void onClick(View v) {
...
                 setupGoogleSignIn();
                 break;
             case R.id.zulip_login:
                 if (!isInputValid()) {
                     return;
                 }
                 commonProgressDialog.showWithMessage(getString(R.string.signing_in));
+                commonProgressDialog.setCancelable(false);
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/LoginActivity.java]: setCancelable here has no effect as you are calling after showWithMessage

File: app/src/main/java/com/zulip/android/util/CommonProgressDialog.java (hunk 1 above)
Comment (DravitLochan) [inline app/src/main/java/com/zulip/android/util/CommonProgressDialog.java]: done!

File: app/src/main/java/com/zulip/android/util/CommonProgressDialog.java (hunk 2 above)
Comment (DravitLochan) [inline app/src/main/java/com/zulip/android/util/CommonProgressDialog.java]: done

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java (hunk 4 above)
Comment (DravitLochan) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: when I removed setCancelable() from the constructor and deployed the app on my phone, it had it's effect everywhere I added setCancelable after showWithMessage().

**Pull request closed without merging**